Remotion/Web/Development.WebTesting/WebTestSetUpFixtureHelper.cs

```diff
@@ -107,9 +117,47 @@ private void HostWebApplication ()
       _hostingStrategy.DeployAndStartWebApplication();
     }
 
+    private void VerifyWebApplicationStarted (string webApplicationRoot, TimeSpan applicationPingTimeout)
+    {
+      var stopwatch = Stopwatch.StartNew();
+      var webRequest = (HttpWebRequest) HttpWebRequest.Create (webApplicationRoot);
+      webRequest.Method = WebRequestMethods.Http.Head;
+      webRequest.AllowAutoRedirect = true;
+      HttpStatusCode statusCode = default;
+
+      while (statusCode != HttpStatusCode.OK)
+      {
+        try
+        {
+          var response = (HttpWebResponse) webRequest.GetResponse();
+          statusCode = response.StatusCode;
+        }
+        catch (WebException ex)
+        {
+          if (stopwatch.ElapsedMilliseconds > applicationPingTimeout.TotalMilliseconds)
+          {
+            throw new InvalidOperationException (
+                $"Checking the Web Application Root '{webApplicationRoot}' did not return '{HttpStatusCode.OK}' in the defined {nameof (applicationPingTimeout)} '{applicationPingTimeout}'."
+                + $"Failed with: '{ex.Message}'. Abort.");
+          }
+
+          s_log.Warn ($"Checking the Web Application Root '{webApplicationRoot}' failed with '{ex.Message}'. Trying again.");
+        }
+
+        if (stopwatch.ElapsedMilliseconds > applicationPingTimeout.TotalMilliseconds)
+        {
+          throw new InvalidOperationException (
+                $"Checking the Web Application Root '{webApplicationRoot}' did not return '{HttpStatusCode.OK}' in the defined {nameof (applicationPingTimeout)} '{applicationPingTimeout}'."

```

Comment:
lowercase

Reply:
What is the exception message supposed to read when formatted? Place show example.

Reply:
Exception Message after refactoring: 'Checking the web application root 'http://localhost:60401/' did not return HttpStatusCode 'OK' in the defined applicationPingTimeout (00:00:10). Failed with HttpStatusCode 'NotFound'.'